src/Propel/Generator/Builder/Om/ObjectBuilder.php

```diff
@@ -282,6 +282,14 @@ protected function addClassBody(&$script)
             '\Propel\Runtime\Map\TableMap'
         );
 
+        $baseClass = $this->getBaseClass();
+        if (strrpos($baseClass, '\\') !== FALSE) {
+            $classesToDeclare[] = $baseClass;
+        }
+
+        $reflectionMethod = new \ReflectionMethod($this, 'declareClasses');
```

Comment:
This comes down to the preferred programming style. My preference is an object orientated approach. Of course `call_user_func_array` is valid solution, as well.